drivers/regulator/regulator_fixed.c

```diff
@@ -86,21 +86,30 @@ static int regulator_fixed_init(const struct device *dev)
 	const struct regulator_fixed_config *cfg = dev->config;
 
 	regulator_common_data_init(dev);
+	bool is_enabled = false;
 
 	if (cfg->enable.port != NULL) {
 		if (!gpio_is_ready_dt(&cfg->enable)) {
 			LOG_ERR("GPIO port: %s not ready", cfg->enable.port->name);
 			return -ENODEV;
 		}
 
-		int ret = gpio_pin_configure_dt(&cfg->enable, GPIO_OUTPUT);
+		int ret = gpio_pin_configure_dt(&cfg->enable, GPIO_OUTPUT_INACTIVE);
```

Comment:
This should not be set to inactive by default. If it is, you risk a momentary disabling of a regulator which may previously have been enabled by e.g. bootloader. is_enabled should reflect the state that the GPIO is currently is. regulator_common_init will then set the GPIO active or inactive if `regulator-boot-on` or `regulator-boot-off` is set. If neither property is set, the GPIO remains in the state it was already configured as

```suggestion
		int ret = gpio_pin_configure_dt(&cfg->enable, GPIO_OUTPUT);
```